src/operator/controllers/intents_reconcilers/network_policy.go

```diff
@@ -76,6 +77,12 @@ func (r *NetworkPolicyReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 	logrus.Infof("Reconciling network policies for service %s in namespace %s",
 		intents.Spec.Service.Name, req.Namespace)
 
+	intents.Status.UpToDate = false
+
+	defer func() {
+		intents.Status.UpToDate = true
```

Comment:
Let's move this to `intents_controller.go`, which has a `Reconcile` func that calls all the sub-reconcilers. Once they have all succeeded, you can set `UpToDate = true` there, and won't have to duplicate this logic across multiple reconcilers.

Reply:
he method there has no awareness of the underlying fields in the structs. i wrote the code in all files because the only place where i can access the struct fields is after i call `err := r.client.Get(ctx, req.NamespacedName, intents)` . or do you think we should make this call in the Reconcile method in the intents_controller file as well ?

Reply:
Hi there @aerosouund! Jumping in per @orishoshan's request. 
I agree with Ori here - we should avoid duplicating this logic across the different group reconcilers, and instead apply the status update in a single place. This also prevents a possible bug in your current implementation, where the value of UpToDate is ultimately determined only according to the last reconciler execution. 
Indeed, to do so you'd have to call `r.client.Get(...)` at the root reconciler, that is in `IntentsReconciler.Reconcile` function, at intents_reconciller.go. 

If you have any follow-up questions feel free to contact me using comments here, or at our Otterize Community slack.

Reply:
Updated the logic, thanks for the concise explanation